fastchat/serve/gradio_web_server.py

```diff
@@ -725,6 +732,9 @@ def build_single_model_ui(models, add_promotion_links=False):
 
     # Register listeners
     imagebox = gr.State(None)
+    apply_rate_limit = gr.State(True)
```

Comment:
sorry why do we need add a gradio state for rate limit?

Reply:
I wasn't sure how to best represent a default state in gradio. In gradio_web_server (direct chat) when we call `bot_response`, we want `apply_rate_limit` to be true and `use_recommended_config` to be false, but the only way to set this is to pass these parameters into `bot_response` through gradio states. I'm not sure if there is a better way to pass it in.

One fix to bypass this is to make `use_recommended_config` default to False and only set it to `true` in `gradio_web_server_anony`.